x-pack/plugins/security_solution/server/endpoint/services/feature_usage/service.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LicensingPluginSetup, LicensingPluginStart } from '../../../../../licensing/server';
+
+const FEATURES = Object.freeze({
+  HOST_ISOLATION: 'Host isolation',
+  HOST_ISOLATION_EXCEPTION: 'Host isolation exception',
+  HOST_ISOLATION_EXCEPTION_BY_POLICY: 'Host isolation exception by policy',
+  TRUST_APP_BY_POLICY: 'Trusted app by policy',
+  EVENT_FILTERS_BY_POLICY: 'Event filters by policy',
+  RANSOMWARE_PROTECTION: 'Ransomeware protection',
+  MEMORY_THREAT_PROTECTION: 'Memory threat protection',
+  BEHAVIOR_PROTECTION: 'Behavior protection',
+});
+
+export class FeatureUsageService {
+  private licensingPluginStart: LicensingPluginStart | undefined;
+
+  private get notify(): (featureName: string, usedAt?: number | Date | undefined) => void {
+    return this.licensingPluginStart?.featureUsage.notifyUsage || function () {};
+  }
+
+  public setup(licensingPluginSetup: LicensingPluginSetup): void {
+    Object.values(FEATURES).map((featureValue) =>
+      licensingPluginSetup.featureUsage.register(featureValue, 'platinum')
+    );
+  }
+
+  public start(licensingPluginStart: LicensingPluginStart): void {
+    this.licensingPluginStart = licensingPluginStart;
+  }
+
+  public notifyUsage(featureKey: keyof typeof FEATURES): void {
+    this.notify(FEATURES[featureKey]);
```

Comment:
Just to make sure - you are not using the second argument of the function (`usedAt`), right?

Reply:
yeah, it's unused. we're just using the default of now.